Courtesy of Eldad Zack <eldad@gentoo.org>
This patch will disable checksumming when there is no address specified.
http://bugs.gentoo.org/show_bug.cgi?id=158851

--- traceroute.c.orig	2007-08-25 00:27:06.000000000 +0300
+++ traceroute.c	2007-08-25 00:30:17.000000000 +0300
@@ -577,6 +577,9 @@
 
 	if (!doipcksum)
 		Fprintf(stderr, "%s: Warning: ip checksums disabled\n", prog);
+		
+	if (source == NULL)
+		doipcksum = (doipcksum == 0);
 
 	if (lsrr > 0)
 		optlen = (lsrr + 1) * sizeof(gwlist[0]);
